### ISSUE TITLE
✨ Add filters to 'Screening and assessment' page

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#9948]
## User story
As a pool operator, I want to filter the screening and assessment table, so I can focus on (for example) just EE candidates first.

As a pool operator, I want to sort candidates by applications date, so I can assess them in the order that they applied.

## Details
Add filter button to the screening and assessment page.

## Design
@Jerryescandon 
https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Admin---Processes%2FPools%2FRecruitments---Record-of-Decision?type=design&node-id=8738%3A33892&mode=design&t=55lrreznUEbI4ioU-1

##  Screenshots
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/2494551/27f138e7-1641-4bac-89bf-c57e9aa4323c)
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/7dc117aa-5399-4b2a-af54-aeae3f018db3)

## Acceptance criteria

> [!IMPORTANT]
> Sorting is not part of the scope for this ticket. It may be  introduced in the future but for now, do not include it.

- [ ] Filter ~~and sort~~ button added to Screening and assessment page
- [ ] Filter ~~and sort~~ dialogue has all the same options as the designs (ok if they are implemented a little different though)
- [ ] Filtering ~~and sort~~ actually work on the Screening and assessment kanban thingy

🛑 Blockers
```[tasklist]
### Blocked by
- [ ] #8966
```

**Comment by tristan-orourke** [comment]
We can leverage the existing search implementation, we just need to allow this page to work with an API endpoint which is paginated and has an upper limit on the number of candidates it returns at a time. We've talked about implementing an "infinite scroll" on each of the assessment step columns, or a "load more" button. We should probably figure that out first.

In addition, we might want to block this on #8966, and incorporate that into the query first.

**Comment by gobyrne** [comment]
> We've talked about implementing an "infinite scroll" on each of the assessment step columns, or a "load more" button. We should probably figure that out first.

@tristan-orourke do you have an issue for this yet?

**Comment by vd1992** [comment]
> > We've talked about implementing an "infinite scroll" on each of the assessment step columns, or a "load more" button. We should probably figure that out first.
> 
> @tristan-orourke do you have an issue for this yet?

Friendly reminder ping @tristan-orourke 
Will there be another issue to be created that will block this one

**Comment by brindasasi** [comment]
> > > We've talked about implementing an "infinite scroll" on each of the assessment step columns, or a "load more" button. We should probably figure that out first.
> > 
> > 
> > [@tristan-orourke](https://github.com/tristan-orourke) do you have an issue for this yet?
> 
> Friendly reminder ping [@tristan-orourke](https://github.com/tristan-orourke) Will there be another issue to be created that will block this one

IMO, it can be implemented under this story itself. The api is there and its just implementing load more in here. Any preference on infinite scroll vs load more in terms of accessiblility?

**Comment by brindasasi** [comment]
Decision: We don't need to change current implementation in this story. We could just add filtering & sorting in here.

**Comment by brindasasi** [comment]
We don't need sorting in this story .. @esizer please help adjusting A/C
